src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -572,6 +956,114 @@ private function compare(mixed $left, mixed $right, string $operator): bool
         };
     }
 
+    /**
+     * Only values of the same type can be compared for ordering, apart from integer and float types.
+     */
+    private function areComparableTypes(mixed $left, mixed $right): bool
+    {
+        $leftType = \gettype($left);
+        $rightType = \gettype($right);
+
+        return $leftType === $rightType || (('integer' === $leftType || 'double' === $leftType) && ('integer' === $rightType || 'double' === $rightType));
+    }
+
+    private function isNonSingularQuery(string $expr): bool
+    {
+        try {
+            $tokens = JsonPathTokenizer::tokenize(new JsonPath($expr));
+
+            foreach ($tokens as $token) {
+                if (TokenType::Bracket === $token->type) {
+                    $trimmedValue = trim($token->value);
+
+                    if (
+                        str_contains($token->value, ',')
+                        || '*' === $trimmedValue
+                        || preg_match('/^(-?\d*+)\s*+:\s*+(-?\d*+)(?:\s*+:\s*+(-?\d*+))?$/', $trimmedValue)
+                    ) {
+                        return true;
+                    }
+                }
+
+                if (TokenType::Name === $token->type && '*' === $token->value || TokenType::Recursive === $token->type) {
+                    return true;
+                }
+            }
+
+            return false;
+        } catch (\Throwable) {
```

Comment:
please review the PR and remove all silencing catch-all - things should be made specific